src/dodal/devices/aperturescatterguard.py

```diff
@@ -119,7 +121,9 @@ def __init__(
             ],
         )
         with self.add_children_as_readables(HintedSignal):
-            self.selected_aperture = soft_signal_rw(ApertureValue)
+            self.selected_aperture = create_hardware_backed_soft_signal(
+                ApertureValue, self.get_current_aperture_position
```

Comment:
nit: `get_current_aperture_position` should be made private on the device class (I should have done this in the prior change): its value is currently exposed through device.read() [intended design] and would allow future changes to the method if required.

nit because I don't know how simple the change would be in mx-bluesky and I don't want to hold up broken device